### PR TITLE
front:rollingstockeditor: refacto, remove react-hook-forms and ajv resolver

### LINKED ISSUE #4553: Refactor rolling stock editor
# Description

Replace the input form generated from a schema by a hand made form that uses SNCF inputs.

# Goal

Simplify the code of the RS editor

# AC

- [ ] react-hook-forms and AJV-resolver has been removed
- [ ] the custom openAPI schema has been removed
- [ ] openapi-schema-to-json-schema has been removed
- [ ] The SNCF input style have been used
- [ ] The behaviors of the components are the same (mandatory or not, min/max values, etc)

### PR DESCRIPTION
**Pull request by Yohh**
- use InputSNCF, InputGroupSNCF and SelectSNCF components
- adapt form errors

close #4553 

:warning: : UX/UI refacto will comme with #4556

### CODE REVIEW

File: front/public/locales/en/rollingstock.json [hunk 1]
```diff
@@ -22,10 +22,11 @@
   "electricOnly": "Electric only",
   "equipement": "Equipment",
   "errorMessages": {
-    "missingInformation": "Missing information",
-    "must be number": "Number is required",
-    "must be equal to one of the allowed values": "Authorised value required",
-    "rollingStockUsed": "Rolling stock is used in the following scenarios:",
+    "missingInformation": "missing informations",
+    "must be number": "must be a number",
```
Comment (clarani) [inline front/public/locales/en/rollingstock.json]: We use camelCase for the keys (same for the french file)
```suggestion
    "mustBeNumber": "must be a number",
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 2]
```diff
@@ -128,11 +31,16 @@ const RollingStockEditorForm = ({
   const { t } = useTranslation(['rollingstock', 'translation']);
   const { openModal } = useModal();
   const dispatch = useDispatch();
-  const jsonSchema = openapiSchemaToJsonSchema(RollingStockBaseOpenapiSchema);
-  const schema = jsonSchema.components.schemas.RollingStockBase;
   const [postRollingstock] = osrdEditoastApi.usePostRollingStockMutation();
   const [patchRollingStock] = osrdEditoastApi.usePatchRollingStockByIdMutation();
 
+  const RollingStockschemaProperties = Object.values(schema) as SchemaProperty[];
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]:
```suggestion
```
It would be nicer to transform your Json file into a ts file and to import directetly the schema from the ts file (see comment on the json file)

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 3]
```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
...
-      raise_pantograph_time: rollingStockData?.raise_pantograph_time || null,
-      default_mode: rollingStockData?.effort_curves.default_mode || selectedMode,
-      effort_curves: {
-        modes: {
-          [`${selectedMode}`]: {
-            curves: defaultRollingstockMode.modes[`${selectedMode}`].curves,
-            is_electric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
-            default_curve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
-          },
+  const defaultValues: RollingStockParametersValues = {
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Can you use a useMemo here to avoid instantiating this const at each re-render ? Also, to reduce the number of lines of this file, maybe it will be useful to create a file utils with a default rollingStock.

```suggestion
  // in utils
  export const rollingStockEditorDefaultValues = {
    railjsonVersion: '',
    name: '',
    ...
  };
  
  // in RollingStockEditorForm
  const defaultValues: RollingStockParametersValues = useMemo(() => {
    if (rollingStockData) {
      const modes = { [`${selectedMode}`]: {
          curves: defaultRollingstockMode.modes[`${selectedMode}`].curves,
          isElectric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
          defaultCurve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
        } };
        return { ...rollingStockData, effortCurves: { modes } };
    }
    return rollingStockEditorDefaultValues;
  }, [rollingStockData, selectedMode, defaultRollingStockMode])
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 4]
```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
...
+          isElectric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
+          defaultCurve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
         },
       },
     },
-  } as UseFormProps<RollingStockParametersValues>);
+  };
+
+  const [rollingStockValues, setRollingStockValues] =
+    useState<RollingStockParametersValues>(defaultValues);
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]:
```suggestion
    useState(defaultValues);
```
Not needed I think :)

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 5]
```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
...
-  // the array is split in two to display the metadatas in two columns
-  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
-    Object.keys(schema.properties)
-      .filter(
-        (property, index) =>
-          property === Metadata[property as Metadata] && (position ? index <= 5 : index > 5)
-      )
+  const metadataForm = (data: SchemaProperty[]) =>
+    data
+      .filter((property) => property.title === Metadata[property.title as Metadata])
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Transform Metadata in an array and use includes. You can also rename it RollingStockMetadataFields
```suggestion
      .filter((property) => RollingStockMetadataFields.includes(property.title))
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 6]
```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
...
+            sm
+            isFlex
+            key={property.title}
+          />
+        );
+      });
 
-  const queryArg = (data: FieldValues): RollingStockUpsertPayload => ({
-    railjson_version: data.railjson_version,
+  const queryArg = (data: RollingStockParametersValues): RollingStockUpsertPayload => ({
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Move this to the utils file, to decrease the number of lines :)

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 7]
```diff
@@ -328,7 +292,7 @@ const RollingStockEditorForm = ({
     },
   });
 
-  const addNewRollingstock = (data: FieldValues) => {
+  const addNewRollingstock = (data: RollingStockParametersValues) => {
     postRollingstock({
       locked: false,
       rollingStockUpsertPayload: queryArg(data),
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: I tested and the catch doesn't work
```suggestion
  const addNewRollingstock = (data: RollingStockParametersValues) => {
    postRollingstock({
      locked: false,
      rollingStockUpsertPayload: queryArg(data),
    })
      .unwrap()
      .then(
        () => {
          dispatch(
            setSuccess({
              title: t('messages.success'),
              text: t('messages.rollingStockAdded'),
            })
          );
          setAddOrEditState(false);
        },
        (error) => {
          if (error.data?.message.includes('duplicate')) {
            dispatch(
              setFailure({
                name: t('messages.failure'),
                message: t('messages.rollingStockDuplicateName'),
              })
            );
          } else {
            dispatch(
              setFailure({
                name: t('messages.failure'),
                message: t('messages.rollingStockNotAdded'),
              })
            );
          }
        }
      )
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 8]
```diff
@@ -420,25 +385,72 @@ const RollingStockEditorForm = ({
     );
   };
 
-  // TODO: use InputSNCF and InputGroupSNCF
+  useEffect(() => {
+    setIsValid(true);
+    RollingStockschemaProperties.forEach((property) => {
+      if (
+        (property.title === Parameter[property.title as Parameter] &&
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]:
```suggestion
        (RollingStockParameters.includes(property.title) &&
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 9]
```diff
@@ -420,25 +385,72 @@ const RollingStockEditorForm = ({
...
+          Number.isNaN(rollingStockValues[property.title])) ||
+        (rollingStockValues[property.title] as number) < (property.min as number) ||
+        (rollingStockValues[property.title] as number) > (property.max as number)
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx:165]: I don't really understand this test

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 10]
```diff
@@ -420,25 +385,72 @@ const RollingStockEditorForm = ({
...
-      onSubmit={handleSubmit(submit)}
+      onSubmit={(e) => submit(e, rollingStockValues)}
     >
-      <div className="d-lg-flex justify-content-center mb-4">
-        <div className="col-lg-6 mr-4">{metadataForm(sideValue.left, register)}</div>
-        <div className="col-lg-6">{metadataForm(sideValue.right, register)}</div>
+      <div className="d-lg-flex justify-content-center mb-4 px-1">
+        <div className="col-lg-4 mr-4 rollingstock-editor-input-container">
+          {metadataForm(
+            RollingStockschemaProperties.filter((schemaProperty) => schemaProperty.side === 'left')
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: You can remove the filter in metadataForm if you filter all the fields here: 
```suggestion
            RollingStockschemaProperties.filter((schemaProperty) => schemaProperty.side === 'left' && Metadata.includes(schemaProperty.title)
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx [hunk 11]
```diff
@@ -1,14 +1,15 @@
 import React from 'react';
 import { useModal } from 'common/BootstrapSNCF/ModalSNCF';
 import { useTranslation } from 'react-i18next';
-import { FieldValues } from 'react-hook-form';
+// import { FieldValues } from 'react-hook-form';
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx]:
```suggestion
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx [hunk 12]
```diff
@@ -63,7 +64,7 @@ const RollingStockEditorFormModal = ({
             type="button"
             className="btn btn-sm btn-primary ml-3"
             onClick={() => {
-              if (request) request(data as FieldValues);
+              if (request) request(data as RollingStockParametersValues);
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx]:
```suggestion
              if (request && data) request(data);
```

File: front/src/applications/rollingStockEditor/consts.ts [hunk 13]
```diff
@@ -1,47 +1,62 @@
...
+  side: string;
+  format?: string;
+  enum?: string[];
+  min?: number;
+  max?: number;
+  unit?: string;
+  units?: string[];
+};
+
 export enum Metadata {
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/consts.ts]: As Metadata is exported, we can have access to this enum everywhere in the application, and the name is not very precise here. You should also transform this into an array I think

```suggestion
export enum RollingStockEditorMetadata {
```

File: front/src/applications/rollingStockEditor/consts.ts [hunk 14]
```diff
@@ -56,25 +71,14 @@ export enum Metadata {
 export enum Parameter {
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/consts.ts]: Same here, can you rename it more precisely and transform it into an array ?

File: front/src/applications/rollingStockEditor/json/rollingStockEditorData.json [hunk 15]
```diff
@@ -0,0 +1,139 @@
+{
+  "name": {
+    "title": "name",
+    "type": "string",
+    "side": "left"
+  },
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/json/rollingStockEditorData.json]: Transform your json file into a TS one :)

```suggestion
import { SchemaProperty } from '../consts';

const RollingStockEditorSchema: SchemaProperty[] = [
  {
    title: 'name',
    type: 'string',
    side: 'left',
  }, ...
];

export default RollingStockEditorSchema;

```

File: front/public/locales/en/rollingstock.json [hunk 16]
```diff
@@ -22,10 +22,11 @@
...
+    "minMaxRangeError": "the value must be between {{min}} and {{max}}",
```
Comment (clarani) [inline front/public/locales/en/rollingstock.json]: Why did you remove all the capital letters ?

**Review by clarani — CHANGES_REQUESTED** [review]
Nice job ! Left some comments :)

**Comment by clarani** [comment]
image
The number should be displayed with a comma "," and not a "." => 0,01

File: front/public/locales/en/rollingstock.json (hunk 1 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: it was for ajv resolver errors, I have to remove them

File: front/public/locales/en/rollingstock.json (hunk 16 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: I forgot to set capital letters, done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 4 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 7 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: are you sure? I just tested and it worked for me

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 9 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx:165]: if the input is empty
or if the number is less than the minimum
or if the number is more than the maximum

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 8 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: this test is only for parameters, no need of this for metadatas

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx (hunk 11 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx (hunk 12 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx]: doing that makes duplicate and delete impossible

File: front/src/applications/rollingStockEditor/json/rollingStockEditorData.json (hunk 15 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/json/rollingStockEditorData.json]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 6 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 2 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: done

File: front/src/applications/rollingStockEditor/consts.ts (hunk 13 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/consts.ts]: done

File: front/src/applications/rollingStockEditor/consts.ts (hunk 14 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/consts.ts]: done

File: front/public/locales/en/rollingstock.json [hunk 17]
```diff
@@ -22,10 +22,11 @@
...
+    "minRangeError": "the value must be greater than {{min}}",
+    "must be equal to one of the allowed values": "must be equal to one of the allowed values",
```
Comment (Akctarus) [inline front/public/locales/en/rollingstock.json]: use camelCase instead

File: front/public/locales/fr/rollingstock.json [hunk 18]
```diff
@@ -24,21 +24,23 @@
   "errorMessages": {
     "missingInformation": "Informations manquantes",
     "must be number": "Nombre requis",
+    "minMaxRangeError": "la valeur doit être comprise entre {{min}} et {{max}}",
+    "minRangeError": "la valeur doit être supérieure ou égale à {{min}}",
     "must be equal to one of the allowed values": "Valeur autorisée requise",
```
Comment (Akctarus) [inline front/public/locales/fr/rollingstock.json]: Same as above, camelCase must be used

**Review by SharglutDev — CHANGES_REQUESTED** [review]
- Not in the changed files but shouldn't you be fetching the rollingstocks in RollingStockEditor.tsx instead of Home.tsx ?
- I think you need a useEffect in RollingStockEditor.tsx something like :
`  useEffect(() => {
    if (isSuccess && rollingStocks.length > 0) {
      setFilteredRollingStockList(rollingStocks);
    }
  }, [isSuccess]);`
because the list of rollingstocks is not loaded when we enter the editor

**Review by Akctarus — CHANGES_REQUESTED** [review]

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 7 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Tested and yes it works !

File: front/public/locales/en/rollingstock.json (hunk 17 above)
Comment (Yohh) [reply front/public/locales/en/rollingstock.json]: done

File: front/public/locales/fr/rollingstock.json (hunk 18 above)
Comment (Yohh) [reply front/public/locales/fr/rollingstock.json]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx (hunk 12 above)
Comment (Yohh) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx]: done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 19]
```diff
@@ -128,210 +35,181 @@ const RollingStockEditorForm = ({
...
-  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
-    Object.keys(schema.properties)
+  const metadataForm = (data: SchemaProperty[]) =>
+    data
       .filter(
-        (property, index) =>
-          property === Metadata[property as Metadata] && (position ? index <= 5 : index > 5)
+        (property) =>
+          property.title ===
+          RollingStockEditorMetadata[property.title as RollingStockEditorMetadata]
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: At this stage, we can also filter by side. This way, you can remove the filters l332, l338 and l344

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 20]
```diff
@@ -128,210 +35,181 @@ const RollingStockEditorForm = ({
...
-      },
-    },
-  } as UseFormProps<RollingStockParametersValues>);
+  const [rollingStockValues, setRollingStockValues] = useState(defaultValues);
 
-  // in the top of the form we display all the metadatas
-  // the array is split in two to display the metadatas in two columns
-  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
-    Object.keys(schema.properties)
+  const metadataForm = (data: SchemaProperty[]) =>
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: UseMemo here too

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 21]
```diff
@@ -128,210 +35,181 @@ const RollingStockEditorForm = ({
...
-      .filter((property) =>
-        position
-          ? !RollingStockResistance[property as RollingStockResistance]
-          : RollingStockResistance[property as RollingStockResistance]
+  const parameterForm = (data: SchemaProperty[]) =>
+    data
+      .filter(
+        (property) =>
+          property.title ===
+          RollingStockEditorParameter[property.title as RollingStockEditorParameter]
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Add the filter by side here too, also add useMemo

File: front/src/applications/rollingStockEditor/json/rollingStockEditorData.json
Comment (clarani) [inline front/src/applications/rollingStockEditor/json/rollingStockEditorData.json:1]: This file can be deleted

File: front/src/applications/rollingStockEditor/utils.ts [hunk 22]
```diff
@@ -0,0 +1,131 @@
...
+      is_electric: false,
+    },
+  },
+});
+
+const getRollingStockEditorDefaultValues = (
+  selectedMode: string,
+  rollingStockData?: RollingStock
+): RollingStockParametersValues => ({
+  railjsonVersion: rollingStockData?.railjson_version || '',
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/utils.ts]: Suggestion: only do 1 test if rolllingStockData is undefined or not (it increases the number of lines, but I think it's nicer)

File: front/src/applications/rollingStockEditor/consts.ts [hunk 23]
```diff
@@ -53,28 +66,157 @@ export enum Metadata {
...
-  rolling_resistance_A = 'rolling_resistance_A',
-  rolling_resistance_B = 'rolling_resistance_B',
-  rolling_resistance_C = 'rolling_resistance_C',
-}
-
-export enum RollingStockResistance {
-  rolling_resistance_A = 'rolling_resistance_A',
-  rolling_resistance_B = 'rolling_resistance_B',
-  rolling_resistance_C = 'rolling_resistance_C',
+  maxSpeed = 'maxSpeed',
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/consts.ts:72]: Suggestion: no need for the repetition
```suggestion
  maxSpeed,
  ...
```

File: front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorFormModal.tsx:1]: Nice !!

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx [hunk 24]
```diff
@@ -128,210 +35,181 @@ const RollingStockEditorForm = ({
...
   const { openModal } = useModal();
   const dispatch = useDispatch();
-  const jsonSchema = openapiSchemaToJsonSchema(RollingStockBaseOpenapiSchema);
-  const schema = jsonSchema.components.schemas.RollingStockBase;
   const [postRollingstock] = osrdEditoastApi.usePostRollingStockMutation();
   const [patchRollingStock] = osrdEditoastApi.usePatchRollingStockByIdMutation();
 
+  const [isCurrentEffortCurveDefault, setIsCurrentEffortCurveDefault] = useState<boolean>(true);
+  const [isValid, setIsValid] = useState<boolean>(true);
+  const [optionValue, setOptionValue] = useState<string>('');
```
Comment (clarani) [inline front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: No need to specify the types here
```Suggestion
  const [isCurrentEffortCurveDefault, setIsCurrentEffortCurveDefault] = useState(true);
  const [isValid, setIsValid] = useState(true);
  const [optionValue, setOptionValue] = useState('');
```

**Review by Akctarus — APPROVED** [review]
LGTM

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 5 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 19 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 20 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Moved to create a new component

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 21 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Moved to a new component

File: front/src/applications/rollingStockEditor/utils.ts (hunk 22 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/utils.ts]: Done

File: front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx (hunk 24 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx]: Done

File: front/src/applications/rollingStockEditor/json/rollingStockEditorData.json
Comment (clarani) [reply front/src/applications/rollingStockEditor/json/rollingStockEditorData.json:1]: Done

File: front/src/applications/rollingStockEditor/consts.ts (hunk 23 above)
Comment (clarani) [reply front/src/applications/rollingStockEditor/consts.ts:72]: Can't do that, otherwise the enum object is replaced by integers

**Review by clarani — COMMENTED** [review]
LGTM, ready for review

**Review by SharglutDev — APPROVED** [review]

**Review by Math-R — APPROVED** [review]
LGTM

**Review by clarani — APPROVED** [review]
LGTM
